scipy/sparse/csgraph/tests/test_spanning_tree.py

```diff
@@ -64,3 +65,62 @@ def test_minimum_spanning_tree():
 
         npt.assert_array_equal(mintree.toarray(), expected,
             'Incorrect spanning tree found.')
+
+@pytest.mark.parametrize("dtype", [np.int32, np.int64])
+def test_mst_with_various_index_dtypes(dtype):
+    #CSR array
+    # Row indices
+    indptr = np.array([0, 2, 4, 5], dtype=dtype)
+    indices = np.array([1, 2, 0, 2, 1], dtype=dtype)
+    data = np.array([2, 0, 2, 3, 3], dtype=float)
+
+    graph = csr_array((data, indices, indptr), shape=(3, 3))
+
+    # Check whether the dtype of indices is as expected
+    assert graph.indices.dtype == dtype
+    assert graph.indptr.dtype == dtype
+
+    # Compute MST
+    mst = minimum_spanning_tree(graph)
+
+    # Check whether the dtype of indices is as expected
+    assert mst.indices.dtype == dtype
+    assert mst.indptr.dtype == dtype
+
+    # Expected MST has 2 edges: (0->1, weight 2)
+    expected = np.array([
+        [0, 2, 0],
+        [0, 0, 0],
+        [0, 0, 0]
+    ], dtype=float)
+
+    npt.assert_array_almost_equal(mst.toarray(), expected)
+
+    # COO array
+    data = np.array([1.0, 4.0, 3.0, 2.0, 5.0], dtype=float)
+    row = np.array([0, 1, 2, 3, 4], dtype=dtype)
+    col = np.array([1, 2, 3, 4, 5], dtype=dtype)
+
+    # Build COO graph with (9, 9) shape
+    adj = coo_array((data, (row, col)), shape=(9, 9))
+
+    # Check whether the dtype of indices is as expected
+    assert adj.row.dtype == dtype
+    assert adj.col.dtype == dtype
+
+    # Compute MST
+    mst = minimum_spanning_tree(adj)
+
+    # Check whether the dtype of indices is as expected
+    assert mst.indices.dtype == dtype
+    assert mst.indptr.dtype == dtype
+
+    # Expected MST will include all 5 edges since graph is disconnected elsewhere
+    expected = np.zeros((9, 9), dtype=float)
+    expected[0, 1] = 1.0
+    expected[1, 2] = 4.0
+    expected[2, 3] = 3.0
+    expected[3, 4] = 2.0
+    expected[4, 5] = 5.0
+
+    npt.assert_array_almost_equal(mst.toarray(), expected)
```

Comment:
Minor point -- here and above, any objection to using `assert_allclose()` as the docs for this NumPy function indicate is preferred?

I realize we still use old way in older code, but maybe in new test code we can start using the new way.

Reply:
I addressed this review. Now I am using `xp_assert_close` (I guess its most generic and reliable).